Sprint 17 from 16 start

Kolesnikov-Eugene opened the pull request.

Kolesnikov-Eugene committed:
DarkMode
localization
Pin - unpin tracker
filters

Kolesnikov-Eugene committed:
Trackers main screen checkmark button adapted to dark mode

Kolesnikov-Eugene committed:
YandexMetrica added to the project

Kolesnikov-Eugene committed:
worckspace

Kolesnikov-Eugene committed:
Full functionality for 17 sprint.
Refactoring

Kolesnikov-Eugene committed:
Corrected filter search bug

LizaIlinykh commented on Tracker/Domain/CategoriesManager.swift:
```diff
@@ -0,0 +1,86 @@
+    }
+    
+    private let context: NSManagedObjectContext
+    private var dataStore: DataStoreProtocol
```
можно тоже сделать константой

Kolesnikov-Eugene committed:
Fixed contentView height in NewHabitViewController.
Corrected title for editing in NewHabitViewController.
TabBar border appearance changed for Dark mode.
Cascade deletion of trackersCompletionRecords in DB when category or tracker is deleted.

Kolesnikov-Eugene committed:
final ver
